hack/verify-deps.sh

```
@@ -0,0 +1,92 @@
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+PROJECT_ROOT="$(cd "${SCRIPT_DIR}/.." && pwd)"
+
+cd "${PROJECT_ROOT}"
+
+echo "Verifying go.mod dependencies are consistent across all modules..."
+
+# Store original files to compare
+TMPDIR=$(mktemp -d)
+trap "rm -rf ${TMPDIR}" EXIT
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)